ms2query/spectrum_processing.py

```diff
@@ -3,33 +3,52 @@
 into account here. Processing here hence refers to inspecting, filtering,
 adjusting the spectrum peaks (m/z and intensities).
 """
-from typing import Dict, Union
+from typing import Dict, Union, List
 import numpy as np
+from matchms import Spectrum
+from spec2vec import SpectrumDocument
+from tqdm import tqdm
 from matchms.typing import SpectrumType
-from matchms.filtering import normalize_intensities, \
-    select_by_mz, select_by_intensity, reduce_to_number_of_peaks, add_losses
+from matchms.filtering import normalize_intensities, select_by_mz, \
+    select_by_intensity, reduce_to_number_of_peaks, add_losses
+from matchms.filtering.require_precursor_mz import require_precursor_mz
 
 
-def set_minimal_processing_defaults(**settings: Union[int, float]
-                                    ) -> Dict[str, Union[int, float]]:
-    """Set default argument values (where no user input is given).
+def minimal_processing_multiple_spectra(spectrum_list: List[SpectrumType],
+                                        progress_bar: bool = False,
```

Comment:
Many functions have the argument `progress_bar`. From the runs that you already did: Would you rather switch if on/off once or per function that you call? In the first case we could just set it once when the highest-level wrapper function is called and then pass it on via `self.progress_bar`.

Reply:
Most of the time I want to do it all together on or altogether of. For create_sqlite_database.py, I already did this in this manner, but I did not yet do this for MS2Library. For spectrum_processing.py I think it is already applied in a good way. I changed the use of progress_bar in ms2library, so this is now set with self.progress_bar.

Reply:
Nice! That should be fine for now.